ch13.md

```diff
@@ -0,0 +1,332 @@
+
+Let's use our curried `reduce` function and make a safe version where the `empty` value is not optional. It shall henceforth be known as `fold`:
+
+```js
+// fold :: Monoid m => m -> [m] -> m
+const fold = reduce(concat)
+```
+
+The initial `m` is our `empty` value - our neutral, starting point, then we take an array of `m`'s and crush them down to one beautiful diamond like value.
+
+```js
+fold(Sum.empty(), [Sum(1), Sum(2)]) // Sum(3)
+fold(Sum.empty(), []) // Sum(0)
+
+fold(Any.empty(), [Any(false), Any(true)]) // Any(true)
+fold(Any.empty(), []) // Any(false)
+
+
+fold(Either.of(Max.empty()), [Right(Max(3)), Right(Max(21)), Right(Max(11))]) // Right(Max(21))
+fold(Either.of(Max.empty()), [Right(Max(3)), Left('error retrieving value'), Right(Max(11))]) // Left('error retrieving value')
+
+fold(IO.of([]), ['.link', 'a'].map($)) // IO([<a>, <button class="link"/>, <a>])
+```
+
+We've provided a manual `empty` value for those last two since we can't define one on the type itself. That's totally fine. Typed languages can figure that out by themselves, but we have to pass it in here.
+
+## Not quite a monoid
+
+There are some *semigroups* that cannot become *monoids*, that is provide an initial value. Look at `First`:
+
+```js
+const First = x => ({ x, concat: other => First(x) })
+
+Map({id: First(123), isPaid: Any(true), points: Sum(13)}).concat(Map({id: First(2242), isPaid: Any(false), points: Sum(1)}))
+// Map({id: First(123), isPaid: Any(true), points: Sum(14)})
+```
+
+We'll merge a couple of accounts and keep the `First` id. There is no way to define an `empty` value for it. Doesn't mean it's not useful.
+
+
+## Grand unifying theory
+
+## Group theory or Category theory?
+
+The notion of a binary operation is everywhere in abstract algebra. It is, in fact, the primary operation for a *category*. We cannot, however, model our operation in catgegory theory without an *identity*. This is the reason we start with a semi-group from group theory, then jump to a monoid in category theory once we have *empty*.
+
+Monoids form a single object category where the morphism is `concat`, `empty` is the identity, and composition is guaranteed.
+
+### Composition as a monoid
+
+Functions of type `a -> a`, where the domain is in the same set as the codomain, are called *endomorphisms*. We can make a *monoid* called `Endo` which captures this idea:
+
+```js
+const Endo = run => ({
+  run,
+  concat: other =>
+    Endo(compose(run, other.run))
+})
+
+Endo.empty = () => Endo(identity)
+
+
+// in action
+
+// thingDownFlipAndReverse :: Endo [String] -> [String]
+const thingDownFlipAndReverse = fold(Endo(() => []), [Endo(reverse), Endo(sort), Endo(append('thing down')])
```

Comment:
Shouldn't it be `fold(Endo(identity), [...])` here since we are folding on endomorphisms. Here we're just going to override the inputs given later to `[]` and apply every next functions to this empty array ^^"